### ISSUE TITLE
Processing Fuse profile file fails

### ISSUE THREAD

**Issue by mrcook** [skoolkid/skoolkit#102]
I've created a profile from FUSE of the Cyclone game, but sna2skool fails when processing it. I've tried this process on both macOS 10.11.6 (python 3.6.3) and Ubuntu 14.04 (python 3.4.3), and both fail with the same message.

The latest skoolkit is cloned and installed as per your documentation page.

The Cyclone tape is loaded into FUSE (v1.4.1), then at the menu screen I save a z80 snapshot. I start the profiler, play through a large part of the game, and save it. The following command is then run:

`sna2skool.py -M cyclone.profile -g cyclone.ctl cyclone.z80 > cyclone.skool`

which produces the following error:

```
Reading cyclone.profile: 100%
Traceback (most recent call last):
  File "/usr/local/bin/sna2skool.py", line 24, in <module>
    sna2skool.main(sys.argv[1:])
  File "/usr/local/lib/python3.6/site-packages/skoolkit/sna2skool.py", line 141, in main
    run(snafile, namespace, config)
  File "/usr/local/lib/python3.6/site-packages/skoolkit/sna2skool.py", line 56, in run
    ctls = generate_ctls(snapshot, start, end, options.code_map)
  File "/usr/local/lib/python3.6/site-packages/skoolkit/snaskool.py", line 537, in generate_ctls
    ctls = _generate_ctls_with_code_map(snapshot, start, end, code_map)
  File "/usr/local/lib/python3.6/site-packages/skoolkit/snaskool.py", line 244, in _generate_ctls_with_code_map
    if _is_terminal_instruction(disassembler.disassemble(b_start, b_end)[-1]):
  File "/usr/local/lib/python3.6/site-packages/skoolkit/snaskool.py", line 188, in _is_terminal_instruction
    if data[0] == 24 and data[1] > 0:
IndexError: list index out of range
```

this is the same error produced on both macOS and Ubuntu.

I also tried a much small `.profile` (a couple of seconds on the Cyclone menu screen) with the same error. Here's the short profile:

```
0x028e,10444
0x0290,14920
0x0293,14920
0x0296,154636
0x0298,47744
0x0299,83590
0x029b,143251
0x02ab,47744
0x02ac,95526
0x02ae,135791
0x02b0,5987
0x02b1,5968
0x02b2,16412
0x031e,5968
0x031f,10444
0x0321,5968
0x0322,10444
0x0324,16412
0x83c2,808
0x83c3,2222
0x83c4,3030
0x83c6,2222
0x83c7,2222
0x83c8,2222
0x83c9,2626
0x83cc,1414
0x83ce,2020
0x842d,2020
0x842e,2020
0x842f,2020
0x8430,2828
0x8432,2020
0x8433,808
0x8434,2020
0x8b74,598649
0x8b78,105847
0x8b79,185237
0x8b7b,105885
0x8b7c,105942
0x8b7d,396950
0x8b7f,185237
0x8b81,396912
0x8b83,185256
0x8b85,105847
0x8b86,185294
0x8b88,397026
0x8b8a,317307
0x8b8c,258
0x8b8d,498536
0x8b90,264627
0xe29e,25383
0xe2a1,25364
0xe2a4,17904
0xe3f8,25383
0xe3fb,10463
0xe3fd,25402
0xe400,17904
0xe40c,5968
0xe40d,5968
0xe40e,14920
0xfe92,14939
0xfe95,10444
0xfe97,623426
0xfe98,396698
0xfe99,396755
0xfe9b,642814
0xfe9d,74569
0xfe9f,82024
0xfea0,44749
0xfea1,52204
0xfea2,44749
0xfea3,52185
0xfea4,52223
0xfea5,82005
0xfea6,74588
0xfea9,74588
0xfeac,52185
0xfeae,117258
0xfeaf,154476
0xfeb1,55747
0xfeb2,55747
0xfeb3,120801
0xfeb9,52185
0xfeba,74550
0xfebb,74569
0xfebc,52204
0xfebd,340048
0xfebe,340048
0xfebf,340105
0xfec0,566810
0xfec1,729379
0xfec3,10437
0xfec5,82125
0xfec6,449902
0xfec9,105866
0xfeca,185218
0xfecc,278213
0xfece,31508
0xfecf,31508
0xfed0,63035
0xfed2,31508
0xfed3,31508
0xfed4,31508
0xfed5,78789
0xfed8,31508
0xfed9,55158
0xfedb,86666
0xfedc,55139
0xfedd,55139
0xfedf,57243
0xfee1,74619
0xfee3,44779
0xfee4,52239
0xfee5,44779
0xfee6,52220
0xfee7,52220
0xfee8,82098
0xfee9,74600
0xfeec,74619
0xfeef,52220
0xfef1,111709
0xfef2,144989
0xfef4,55782
0xfef5,55839
0xfef6,116860
0xfef8,52220
0xfef9,74638
0xfefa,52220
0xfefb,74600
0xfefc,89520
0xfefe,14920
0xfff4,2020
0xffff,2424
```

**Comment by skoolkid** [comment]
Thanks for reporting this bug!

The final line of the profile (0xffff,2424) looks suspicious to me. Is the final byte of the snapshot 24, by any chance? You can check by doing:

    $ ./snapinfo.py -p 65535 cyclone.z80

If it is 24 (18 hex), then as a workaround you could try changing it to 0:

    $ ./snapmod.py -f -p 65535,0 cyclone.z80

And then rerun sna2skool.py. If that doesn't fix the problem, please e-mail me the snapshot (cyclone.z80) and I'll take a closer look.

In any case, I'll get this fixed for 6.3.

**Comment by mrcook** [comment]
Indeed it is 24! Thanks for the help Richard, it's very much appreciated.

**Comment by skoolkid** [comment]
Fixed in 580ad11acea66bc37d3033cd6b488fad9affe74b.
